src/Microsoft.OData.Core/UriParser/Parsers/FunctionOverloadResolver.cs

```diff
@@ -163,42 +164,43 @@ internal static bool ResolveOperationFromList(string identifier, IEnumerable<str
                 throw;
             }
 
-            IList<IEdmAction> foundActionsWhenLookingForFunctions = new List<IEdmAction>();
+            IList<IEdmOperation> foundActionsWhenLookingForFunctions = new List<IEdmOperation>();
             int parameterNamesCount = parameterNames.Count();

```

Comment:
>int parameterNamesCount = parameterNames.Count() [](start = 12, length = 48)

would it be better to make this a bool hasParameters = (parameterNames.FirstOrDefault == null) ?

Reply:
parameterNames is IList<string>, so, Count< T > is cheaper than FirstOrDefault == null ?

---
In reply to: [290003826](https://github.com/OData/odata.net/pull/1473#discussion_r290003826) [](ancestors = 290003826)

Reply:
still probably better to make it bool hasParameters = (parameterNames.Count() > 0), since the only thing we care about is whether there are any parameters (we don't care how many), so that saves a comparison later.

---
In reply to: [290007315](https://github.com/OData/odata.net/pull/1473#discussion_r290007315) [](ancestors = 290007315,290003826)